tmail-backend/blob/blobid-list/src/test/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreContract.scala

```diff
@@ -19,21 +21,53 @@ trait SingleSaveBlobStoreContract extends BlobStoreDAOContract {
   def defaultBucketName: BucketName
 
   @Test
-  def saveShouldSuccessIfBlobIdDoesNotExist(): Unit = {
+  def saveBytesShouldSuccessIfBlobIdDoesNotExist(): Unit = {
     val blobId: BlobId = blobIdFactory.randomId()
     SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block()
 
     assertThat(SMono.fromPublisher(blobIdList.isStored(blobId)).block()).isTrue
   }
 
   @Test
-  def saveShouldNoopIfBlobExists(): Unit = {
+  def saveBytesShouldNoopIfBlobExists(): Unit = {
     val blobId: BlobId = blobIdFactory.randomId()
     SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block()
 
     assertThat(SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block())
   }
 
+  @Test
+  def saveInputStreamShouldSuccessIfBlobIdDoesNotExist(): Unit = {
+    val blobId: BlobId = blobIdFactory.randomId()
+    SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, new ByteArrayInputStream(SHORT_BYTEARRAY))).block()
+
+    assertThat(SMono.fromPublisher(blobIdList.isStored(blobId)).block()).isTrue
+  }
+
+  @Test
+  def saveInputStreamShouldNoopIfBlobExists(): Unit = {
+    val blobId: BlobId = blobIdFactory.randomId()
+    SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, new ByteArrayInputStream(SHORT_BYTEARRAY))).block()
+
+    assertThat(SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, new ByteArrayInputStream(SHORT_BYTEARRAY))).block())
```

Comment:
You are not asserting anything here??? (and actually there is a few other tests that are like that it seems)

Reply:
I don't know how to handler in this case
If blodIdList exists, It still returns `Mono<Void>` (same as when blobIdList isn't exists).
And BlobId is PK.

Reply:
Look in James code, we use for this usually the `assertThatCode` from assertj with the `doesNotThrowAnyException()` method. Something like this:

```
assertThatCode(() -> SMono.fromPublisher(singleSaveBlobStoreDAO.save(TEST_BUCKET_NAME, blobId, new ByteArrayInputStream(SHORT_BYTEARRAY))).block())
            .doesNotThrowAnyException();
```